### ISSUE TITLE
animation style when removing media objects from stage

### ISSUE THREAD

**Issue by helenvarley** [upstage-org/upstage#707]
meta theater asked if it's possible to change the way that avatars and other objects are animated when they are removed from the stage. currently they all spiral. i explained that we can use the opacity slider to first fade them out and then click to remove, but maybe it would be good to be able to choose this animation in the stage management, as we can for speech bubbles.

i imagine this would be an option in the customisation section of stage management, same as speech bubbles and curtain, & would affect all objects on that stage.

**Comment by vickis** [comment]
we used to also have the eye icon which made things seen and unseen - that
could work if the slider and remove seems too slow?

**Comment by helenvarley** [comment]
do you mean for hiding a drawing in the old UpStage? could be useful - it could be under the lightbulb icon, & actually that might help people to understand how the lightbulb functions, as it's easy to think that when you turn the lightbulb off when the item is on stage, that it's no longer visible. the eye icon could make the object hidden for everyone, but players could still see its frame when selected, and find it in the depth tool when not selected.

but for text objects we already have 3 icons there on the right of the frame. i can see that it would fit to have another icon there on everything else, but if we then have 4 icons for the text ... 

if you set the move speed to as fast as possible, then use the opacity slider, this is pretty much the same effect as hiding the object. it's slower, of course, if you are doing other things with the av & need to change sliders & speeds. a single button to hide in one click would be faster.

**Comment by elisadcg** [comment]
The way of objets are animated : i think a more simple fade in or fade out is a good option.

**Comment by helenvarley** [comment]
ok, to summarise. what we would like is an option in the Customisation section of Stage Management, where the animation style and speed of removing graphic objects from the stage can be set, similar to the speech bubble and curtain settings.
i'm just not sure what it should be called - it would apply to avatars & props, and also to videos and streams ... "Graphic object removal" ?? it sounds a bit clunky ... any better ideas?

**Comment by helenvarley** [comment]
summarising this:

- [ ] default way that objects are removed from the stage should be to fade out.
- [ ] in Stage Management > Customisation an option to change this to the spin (current), & any other way that the object might disappear??

it's a very non-urgent issue, but would be good to at least have the default being a fade in & out.

**Comment by gloriajw** [comment]
Check this out, you will find methods to select object exit options.

**Comment by helenvarley** [comment]
where do i find the options? at the moment the objects are now just immediately disappearing.

**Comment by gloriajw** [comment]
stage mgmt/customisations:

<img width="3374" height="1978" alt="Image" src="https://github.com/user-attachments/assets/6f1c3385-b04d-4905-8dda-048a8f2cb8b8" />

**Comment by helenvarley** [comment]
ok, i found it - but it doesn't seem to be working. i've tried setting it to the float away to the top, at a slow speed, and saving. then reloading the stage, looking at it in two different browsers, putting on an avatar and then clicking the X to remove it. it simply vanishes.
i also tried using the remove option in the context menu, but it also just vanished.

HOWEVER: having so many options in the Stage Management section is really not an ideal solution, as this will apply to all objects on that stage. it means that if i select fly away (float to the top) this is how all objects will leave the stage. i might want that for the butterfly, for example, but not for a prop that is a tree. 

if the animation is to be set in Stage Management for the removal of all objects, this is fine but we really only need / want a few generic objects - e.g. vanish and fade. when we want objects to leave by flying away, we can do that by simply moving the object directly on the stage.

if there's a possibility to set the removal style for individual media items then probably the best way would be to have it work the same way that the voices are for avatars: you can set it in the media edit interface of the Studio, as that media item's default, and if you need to change it temporarily on stage then you do it through the context menu. 
BUT this might be getting more complicated than we really need - to manage all of these things in real-time performance is quite a lot. 

so in conclusion: if it's possible, this would be good:

- a setting in Stage Management for the speed of the fade out of media items (vanish isn't necessary as the fastest fade out would be the equivalent to vanishing) - similar to the fade tool for the speech bubble. Fade should be the default, with the default speed in the middle.
- a setting in the media edit interface where a custom removal effect can be set for that specific media item, which would over-ride the general setting for the stage (only for that media item). 

if this isn't possible then just the first one, the setting in Stage Management, can be a simple fade and speed of fade tool.

**Comment by gloriajw** [comment]
Fixed, made this an individual setting: animation type and speed. Pls test

**Comment by helenvarley** [comment]
thanks.
please remove the "stage default" option, as media items can & often are on multiple stages, so then what happens?

the default should be to vanish, as we can make an avatar fade out by reducing the transparency.

**Comment by gloriajw** [comment]
Will do. Also, I think the option to control exit animation should be in the assignment of the media to the stage. It should not be fixed to that particular avatar. I'm going to move this option so that it can differ per stage.

**Comment by helenvarley** [comment]
ok - you mean one setting for all media on that stage? this is probably good, we don't need to get too fancy 😎 
(& it could be something that people don't use very much anyway!)

**Comment by gloriajw** [comment]
exit animations are now temporary, in context menu. Please close this.

**Comment by helenvarley** [comment]
right, they are in the context menu, but please change the default animation to be "Fade Out" (rather than "Disappear") with the speed set to the middle of the slider.
